dvc/repo/ls_url.py

```diff
@@ -0,0 +1,35 @@
+from dvc.exceptions import URLMissingError
+from dvc.fs import get_cloud_fs
+
+
+def parse_external_url(url):
+    fs_cls, fs_config, fs_path = get_cloud_fs(None, url=url)
+    fs = fs_cls(**fs_config)
+    return fs, fs_path
+
+
+def ls_url(url, recursive=False):
+    fs, fs_path = parse_external_url(url)
+    try:
+        info = fs.info(fs_path)
+    except FileNotFoundError:
+        raise URLMissingError(url)
```

Comment:
```suggestion
    except FileNotFoundError:
        raise URLMissingError(url) from exc
```
I understand that this is similar to what is done in `dvc ls`, but I guess it also does this wrap unnecessarily and we should probably use `FileNotFoundError` everywhere. Might be worth it not wrapping it here even though it is inconsistent with `dvc ls`. Up to you.

Reply:
Better to wrap it here, I think, since it's an expected error.

Reply:
@rlamy Right, but we can wrap it with FileNotFoundError with a url instead of inventing our own exception. This is not a public api yet, but it is obvious that we'll need to sort it out sooner or later 🙂